### PR TITLE
Define XMLHttpRequest failure

### LINKED ISSUE #31129: Define XMLHttpRequest "failure" on the responseText MDN page?
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseText

### What specific section or headline is this issue about?

https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseText#value

### What information was incorrect, unhelpful, or incomplete?

"or null if the request failed" is confusing to me because there is no definition of `failed`. Does this refer to a network failure? a 5XX error?

### What did you expect to see?

An explanation that leaves me confident about when `responseText` will be a string and when it will be `null`.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

The linked section of the spec at https://xhr.spec.whatwg.org/#the-responsetext-attribute was interesting, and perhaps hints that `null` would not be associated with a 5XX error because it does not discuss response codes at all, but it was quite hard to read this part of the spec and reason about when `responseText` would be `null` due to the many layers of spec indirection required to even find a mention of `null`.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/xmlhttprequest/responsetext`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseText
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/xmlhttprequest/responsetext/index.md
* Last commit: https://github.com/mdn/content/commit/0a726c0a04ab286873ad91b5ddee478dd938832d
* Document last modified: 2023-11-08T06:06:49.000Z

</details>

### PR DESCRIPTION
**Pull request by JayAgra**
### Description

Attempting to resolve #31129 by specifying a failed request is a non-2XX response code.

### Motivation

Just looking to clarify the confusion found by @maxpblum

### Additional details

MDN response code docs [https://developer.mozilla.org/en-US/docs/Web/HTTP/Status](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)

### Related issues and pull requests

Fixes #31129

### CODE REVIEW

**Review by wbamberg — COMMENTED** [review]
It looks to me like `null` is never the value. I don't see a path [here](https://xhr.spec.whatwg.org/#text-response) where `null` is returned.

And this code:

```js
const xhr = new XMLHttpRequest();
xhr.open("GET", "https://httpbin.org/status/503");

// If specified, responseType must be empty string or "text"
xhr.responseType = "text";

xhr.onloadend = () => {
      console.log(xhr.status);
      console.log(xhr.responseText);
};

xhr.send();
```

...gives me:

```
503
<empty string>
```

**Comment by JayAgra** [comment]
You're absolutely correct, I'll remove the whole statement "or `null` if the request failed (non 2XX status)"

File: files/en-us/web/api/xmlhttprequest/responsetext/index.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ following a request being sent.
 ## Value
 
 A string which contains either the textual data received using the
-`XMLHttpRequest` or `null` if the request failed or
+`XMLHttpRequest` or `null` if the request failed (non 2XX status) or
 `""` if the request has not yet been sent by calling
 {{domxref("XMLHttpRequest.send", "send()")}}.
```
Comment (wbamberg) [inline files/en-us/web/api/xmlhttprequest/responsetext/index.md:20]:
```suggestion
`XMLHttpRequest` or `""` if the request failed or the response has not yet been received.
```

maybe?

Comment (JayAgra) [reply files/en-us/web/api/xmlhttprequest/responsetext/index.md:20]: Yup, that works!

**Review by wbamberg — APPROVED** [review]
👍 thank you @JayAgra !

**Pull request merged**